layer/layer_store.go

```diff
@@ -334,7 +334,10 @@ func (ls *layerStore) get(l ChainID) *roLayer {
 }
 
 func (ls *layerStore) Get(l ChainID) (Layer, error) {
-	layer := ls.get(l)
+	ls.layerL.Lock()
+	defer ls.layerL.Unlock()
```

Comment:
> Do we need to use a defer here? 

That's a good point. We can avoid the defer.

> Does layer.getReference need the layer store lock?

Yes. This is the bug that's being fixed.